### PR TITLE
fix imap_utf7_decode decoding error caused by the Chinese directory

### PR DESCRIPTION
**Pull request by lincome**
fix imap_utf7_decode decoding error caused by the Chinese directory or other special characters

### CODE REVIEW

File: src/PhpImap/Mailbox.php [hunk 1]
```diff
@@ -225,11 +225,13 @@ public function statusMailbox() {
 	 * @param string $pattern
 	 * @return array listing the folders
 	 */
-	public function getListingFolders($pattern = '*') {
+	public function getListingFolders($pattern = '*', $ifImapUtf7Decode = true) {
 		$folders = $this->imap('list', [$this->imapPath, $pattern]) ?: [];
-		foreach($folders as &$folder) {
-			$folder = imap_utf7_decode($folder);
-		}
+		if($ifImapUtf7Decode){
```
Comment (barbushin) [inline src/PhpImap/Mailbox.php]: Please fix formatting.

File: src/PhpImap/Mailbox.php [hunk 2]
```diff
@@ -804,22 +806,23 @@ public function getMailboxes($search = "*") {
...
+	public function imap($methodShortName, $args = [], $prependConnectionAsFirstArg = true, $throwExceptionClass = Exception::class, $ifImapUtf7Encode = true) {
 		if(!is_array($args)) {
 			$args = [$args];
 		}
-		foreach($args as &$arg) {
-			if(is_string($arg)) {
-				$arg = imap_utf7_encode($arg);
-			}
-		}
+		if($ifImapUtf7Encode){
```
Comment (barbushin) [inline src/PhpImap/Mailbox.php]: Fix formatting.

**Comment by barbushin** [comment]
Please check this https://github.com/barbushin/php-imap/issues/286

**Comment by Sebi94nbg** [comment]
Since you've merged #296, I believe, you can close this PR as well @barbushin. :)

**Pull request closed without merging**